File: src/main/java/com/claudio_personagens/bookingcharactersapi/services/ClientService.java
package com.claudio_personagens.bookingcharactersapi.services;

import com.claudio_personagens.bookingcharactersapi.domain.Client;

import java.util.List;
import java.util.Optional;

public interface ClientService {
    Client save(Client client);
    Optional<Client> findByName(String name);
    Optional<Client> findById(String id);
    List<Client> findAll();
}

File: src/main/java/com/claudio_personagens/bookingcharactersapi/services/ClientService.kt
package com.claudio_personagens.bookingcharactersapi.services

import com.claudio_personagens.bookingcharactersapi.base.BaseService
import com.claudio_personagens.bookingcharactersapi.domain.Client
import com.claudio_personagens.bookingcharactersapi.repositories.ClientRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import java.util.*

@Service
interface ClientService : BaseService<Client, UUID> {
    fun findByName(name: String): Optional<Client>
}